core/domain/user_services.py

```diff
@@ -1412,8 +1413,26 @@ def update_profile_picture_data_url(
         profile_picture_data_url: str. New profile picture url to be set.
     """
     user_settings = get_user_settings(user_id, strict=True)
-    user_settings.profile_picture_data_url = profile_picture_data_url
-    save_user_settings(user_settings)
+    username = user_settings.username
+    if username is None:
+        raise utils.ValidationError(
+            'User does not have a valid username, having value None.')
+    # Ruling out the possibility of different types for mypy type checking.
+    assert isinstance(username, str)
+    fs = fs_services.GcsFileSystem(feconf.ENTITY_TYPE_USER, username)
+    filename_png = 'profile_picture.png'
+    png_binary = utils.convert_png_data_url_to_binary(
+        profile_picture_data_url)
+    fs.commit(filename_png, png_binary, mimetype='image/png')
+
+    profile_picture_binary = utils.convert_png_data_url_to_binary(
```

Comment:
Why are we doing the same thing here and above, and calling them two different names (png_binary and profile_picture_binary)?

Reply:
Ah, Updated.